File: pset2/readability.c
#include <stdio.h>
#include <cs50.h>
#include <ctype.h>
#include <string.h>
#include <math.h>
//prototype functions
int count_letters(string text);
int count_words(string text);
int count_sentences(string text);

int main(void)
{
    //get the text
    string text = get_string("Text: ");
    //count the letters words and sentences
    float letters = count_letters(text);
    float words = count_words(text);
    float sentences = count_sentences(text);

    //calculat the average per 100
    float l = 100 / words * letters;
    float s = 100 / words * sentences;
    //find the index
    int index = round(0.0588 * l - 0.296 * s - 15.8);
    //determine and print the grade level
    if (index < 1)
    {
        printf("Before Grade 1\n");
    }
    else if (index >= 16)
    {
        printf("Grade 16+\n");
    }
    else
    {
        printf("Grade %i\n", index);
    }
}

int count_letters(string text)
{
    int letters = 0;
    for (int i = 0, n = strlen(text); i < n; i++)
    {
        //go through each char and see if it is a letter
        bool x = isalpha(text[i]);
        //if it is a letter count it
        if (x == true)
        {
            letters++;
        }
    }
    return letters;
}

int count_words(string text)
{
    //words is equal to spaces plus one
    int words = 1;
    //fgo through each char and see if it is a space
    for (int i = 0, n = strlen(text); i < n; i++)
    {
        bool x = isspace(text[i]);
        //if it is count it
        if (x == true)
        {
            words++;
        }
    }
    return words;
}

int count_sentences(string text)
{
    int sentences = 0;
    //go through and see if each char is a . ! ?
    for (int i = 0, n = strlen(text); i < n; i++)
    {
        //if it is count it
        if (text[i] == 46 || text[i] == 33 || text[i] == 63)
        {
            sentences++;
        }
    }
    return sentences;
}

File: pset6/readability/readability.py
from cs50 import get_string


def main():
    # get text
    text = get_string("Text: ")
    # count letters
    letters = count_letters(text)
    # count words
    words = count_words(text)
    # count sentences
    sentences = count_sentences(text)

    # index
    l = 100 / words * letters
    s = 100 / words * sentences
    index = round(0.0588 * l - 0.296 * s - 15.8)

    if index < 1:
        print("Before Grade 1")
    elif index >= 16:
        print("Grade 16+")
    else:
        print(f"Grade {index}")


def count_letters(text):
    letters = 0
    for i in range(len(text)):
        # only alpha
        if text[i].isalpha() == True:
            letters += 1
    return letters


def count_words(text):
    words = 0
    spaces = 0
    # words = spaces + 1
    for i in range(len(text)):
        if text[i].isspace() == True:
            spaces += 1
    words = spaces + 1
    return words


def count_sentences(text):
    sentences = 0
    # end punctuation
    for i in range(len(text)):
        if text[i] == "." or text[i] == "?" or text[i] == "!":
            sentences += 1
    return sentences


main()
